### DISCUSSION TITLE
Multiple call's of the async execute function in the AsyncCommand

### DISCUSSION THREAD

**Discussion by Avdam** [TheCodeTraveler/AsyncAwaitBestPractices#122] [Q&A]
Hi,

I developed a XamarinForms smartphone app with a save button. The save button Command is binded to a AsyncCommand. Normally and most of the times the async execute is call 1 time after the save button press. But sometimes the execute is called multiple (2 a 5) times after just 1 press. Before calling the execute a busy flag is set to true. The busy flag is part of the condition for CanExecute. I could not reproduce the issue myself. The issue is seen after implementing logging which shows that the execute is called multiple times in a customer situation using an android 11 smartphone
![Capture](https://user-images.githubusercontent.com/11784803/176477064-8b7cb115-dfaf-44d8-bc3d-498412e407a4.PNG)
. See attachment for implementation details.

Is there any explanation for this behaviour?

**Comment by TheCodeTraveler** [comment]
Can you prove a small sample reproduction? Posting a picture of your code is difficult for me to copy/paste it and reproduce your issue.

**Comment by Avdam** [comment]
Hi Brandon, 

Thanks for fast repy.

Attached is all the code around the AsyncCommand.

Unfortunately I could not reproduce the issue myself. 

Also attached the log file the issue occurs on 12 june at 3:21:20 PM (utc)
Example of the issue (5 x called Execute|) line 915, 947, 950, 982 and 985 all within 3 seconds
The stack traces you find in the log are for debugging purposes: The execute results in a CacheRegistration in which I implemented logging of the stack trace.

Best regards,
Arjan
[ToBrandon.zip](https://github.com/brminnick/AsyncAwaitBestPractices/files/9017064/ToBrandon.zip)

**Comment by Avdam** [comment]
Hi Brandon,

Is there any news about this issue?

Best regards,
Arjan

**Reply by TheCodeTraveler** [reply]
Above, you mentioned that you are unable to reproduce the bug:
> Unfortunately I could not reproduce the issue myself.

Are you able to provide a reproduction?
